src/components/input/InputConnected.tsx

```diff
@@ -0,0 +1,33 @@
+import { IReactVaporState } from '../../ReactVapor';
+import { IInputOwnProps, IInputStateProps, IInputDispatchProps, IInputProps, Input } from './Input';
+import { IDispatch, ReduxUtils } from '../../utils/ReduxUtils';
+import { addInput, removeInput, changeInputValue } from './InputActions';
+import { connect } from 'react-redux';
+import { findWhere } from 'underscore';
+
+const mapStateToProps = (state: IReactVaporState, ownProps: IInputOwnProps): IInputStateProps => {
+  const input = findWhere(state.inputs, { id: ownProps.id });
+  return {
+    valid: input && input.valid,
+    value: input && input.value,
+    disabled: input && input.disabled,
+  };
+};
+
+const mapDispatchToProps = (dispatch: IDispatch, ownProps: IInputOwnProps): IInputDispatchProps => ({
+  onRender: (value: string = '', valid = true, disabled = false) => {
+    const validOnMount = ownProps.validateOnMount
+      && ownProps.validate
+      && ownProps.validate(value);
```

Comment:
Aren't you already getting this result from the `valid` parameter? You do the same thing on line 60 of `Input.tsx` where you call `onRender`.

Reply:
done 👍